staging/src/k8s.io/client-go/restmapper/discovery.go

```diff
@@ -145,27 +145,25 @@ func NewDiscoveryRESTMapper(groupResources []*APIGroupResources) meta.RESTMapper
 // GetAPIGroupResources uses the provided discovery client to gather
 // discovery information and populate a slice of APIGroupResources.
 func GetAPIGroupResources(cl discovery.DiscoveryInterface) ([]*APIGroupResources, error) {
-	apiGroups, err := cl.ServerGroups()
+	gs, rs, err := cl.ServerGroupsAndResources()
 	if err != nil {
-		if apiGroups == nil || len(apiGroups.Groups) == 0 {
-			return nil, err
-		}
-		// TODO track the errors and update callers to handle partial errors.
+		return nil, err
```

Comment:
I expected `return rs, err` to propagate partial results

Reply:
how does this change affect our return value in case of partial discovery? it looks like we have a single (in-tree) caller in DeferredDiscoveryRESTMapper... does that need to be updated to handle partial results?

Reply:
I'd prefer not to block this PR on addressing that issue, if possible. The TODO existed here before and should stay, I'd rather we fix that in a separate PR, @liggitt wdyt?

Reply:
> I'd prefer not to block this PR on addressing that issue, if possible. The TODO existed here before and should stay, I'd rather we fix that in a separate PR, @liggitt wdyt?

As long as this function still behaves the same in the face of partial discovery, I'm ok with carrying the TODO forward. My initial reading was that the changes in this PR made this function hard-fail in the face of partial discovery.

Reply:
Yeah, that was the change, let's bring it to the previous state.

Reply:
@liggitt address the partial result case. Added tests to prove that it behaves as before.